src/main/java/net/sacredlabyrinth/phaed/simpleclans/hooks/discord/DiscordHook.java

```diff
@@ -512,9 +543,18 @@ private void updateRole(ClanPlayer clanPlayer, DiscordAction action) {
         }
     }
 
-    private void updatePermissions(@NotNull ClanPlayer clanPlayer, @Nullable Clan clan, DiscordAction action) {
+    private void updateViewPermission(@NotNull ClanPlayer clanPlayer, DiscordAction action) {
+        Clan clan = clanPlayer.getClan();
         Member member = getMember(clanPlayer);
-        if (clan == null || member == null) {
+        if (clan == null) {
+            return;
+        }
+
+        updateViewPermission(member, clan, action);
+    }
+
+    private void updateViewPermission(@Nullable Member member, @NotNull Clan clan, DiscordAction action) {
```

Comment:
Member could be NotNull and the overloaded method removed.

Reply:
:concern: